powertools-e2e-tests/handlers/idempotency-redis/src/main/java/software/amazon/lambda/powertools/e2e/Function.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package software.amazon.lambda.powertools.e2e;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.util.TimeZone;
+import software.amazon.lambda.powertools.idempotency.Idempotency;
+import software.amazon.lambda.powertools.idempotency.IdempotencyConfig;
+import software.amazon.lambda.powertools.idempotency.Idempotent;
+import software.amazon.lambda.powertools.idempotency.redis.RedisPersistenceStore;
+import software.amazon.lambda.powertools.logging.Logging;
+
+
+public class Function implements RequestHandler<Input, String> {
+    public Function() {
+        Idempotency.config().withConfig(
+                        IdempotencyConfig.builder()
+                                .withExpiration(Duration.of(10, ChronoUnit.SECONDS))
+                                .build())
+                .withPersistenceStore(
+                        RedisPersistenceStore.builder()
```

Comment:
I think this is where the user should configure it's redis info (user/pwd)

Reply:
Reading this, I add a +1 on what I wrote above about the JedisClient... we should probably hide this to the user: JedisConfig, JedisClient, ...

I would see something like:
```
RedisPersistenceStore.builder()
     .withHost(System.getenv("REDIS_HOST"))
     .withPort(System.getenv("REDIS_PORT")) // optional, default 6379
     .withUsername(secretsProvider.getValue("REDIS_USERNAME"))
     .withPassword(secretsProvider.getValue("REDIS_PASSWORD"))
     .withSsl(true) // optional, default false
     .build();
```